internal/services/containers/kubernetes_fleet_manager_resource.go

```diff
@@ -88,30 +108,33 @@ func (r KubernetesFleetManagerResource) Create() sdk.ResourceFunc {
 		Timeout: 30 * time.Minute,
 		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
 			client := metadata.Client.ContainerService.V20240401.Fleets
+			subscriptionId := metadata.Client.Account.SubscriptionId
 
-			var config KubernetesFleetManagerResourceSchema
-			if err := metadata.Decode(&config); err != nil {
+			var model KubernetesFleetManagerModel
+			if err := metadata.Decode(&model); err != nil {
 				return fmt.Errorf("decoding: %+v", err)
 			}
 
-			subscriptionId := metadata.Client.Account.SubscriptionId
-
-			id := fleets.NewFleetID(subscriptionId, config.ResourceGroupName, config.Name)
+			id := fleets.NewFleetID(subscriptionId, model.ResourceGroupName, model.Name)
 
 			existing, err := client.Get(ctx, id)
-			if err != nil {
-				if !response.WasNotFound(existing.HttpResponse) {
-					return fmt.Errorf("checking for the presence of an existing %s: %+v", id, err)
-				}
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
 			}
+
 			if !response.WasNotFound(existing.HttpResponse) {
 				return metadata.ResourceRequiresImport(r.ResourceType(), id)
 			}
 
-			var payload fleets.Fleet
-			r.mapKubernetesFleetManagerResourceSchemaToFleet(config, &payload)
+			params := &fleets.Fleet{
```

Comment:
Since this `params` variable is only used by the `CreateOrUpdateThenPoll` function where we immediately dereference the pointer I don't see any reason to make it one

```suggestion
			params := fleets.Fleet{
```